staging/src/k8s.io/apiserver/pkg/server/secure_serving.go

```diff
@@ -189,7 +189,10 @@ func (s *SecureServingInfo) Serve(handler http.Handler, shutdownTimeout time.Dur
 	if s.HTTP2MaxStreamsPerConnection > 0 {
 		http2Options.MaxConcurrentStreams = uint32(s.HTTP2MaxStreamsPerConnection)
 	} else {
-		http2Options.MaxConcurrentStreams = 250
+		// match http2.initialMaxConcurrentStreams used by clients
+		// this makes it so that a malicious client can only open 400 streams before we forcibly close the connection
+		// https://github.com/golang/net/commit/b225e7ca6dde1ef5a5ae5ce922861bda011cfabd
+		http2Options.MaxConcurrentStreams = 100
```

Comment:
did the default change? it was previously 250

Reply:
The default on the server is 250, the default on the client in 100.  IIUC 100 is what more non-go clients use as well, and this is what cloudflare defaults to now.

Reply:
hmmm https://github.com/kubernetes/kubernetes/issues/33557

Reply:
I do not think that issue is relevant here: https://github.com/golang/go/commit/7a622740655bb5fcbd160eb96887032314842e6e